devnet-sdk/devstack/dsl/dsl.go

```diff
@@ -1,17 +1,68 @@
 package dsl
 
 import (
+	"context"
+	"time"
+
 	"github.com/ethereum-optimism/optimism/devnet-sdk/devstack/stack"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
 )
 
+const defaultTimeout = 30 * time.Second
+
+// common provides a set of common values and methods inherited by all DSL structs.
+// These should be kept very minimal.
+// No public methods or fields should be exposed.
+type common struct {
+	// Ctx is the context for test execution.
+	ctx context.Context
+	// log is the component-specific logger instance.
+	log log.Logger
+	// T is a minimal test interface for panic-checks / assertions.
+	t stack.T
+	// Require is a helper around the above T, ready to assert against.
+	require *require.Assertions
+}
+
+// commonWithLog copies the specified common, replacing the log instance.
+// Not an instance method on common to avoid it being inherited to every component that uses common.
+func commonWithLog(c common, log log.Logger) common {
+	return common{
+		ctx:     c.ctx,
+		log:     log,
+		t:       c.t,
+		require: c.require,
+	}
+}
+
 type System struct {
+	common
+	log log.Logger
 	sys stack.System
 }
 
-// TODO(#15137): decide what to expose publicly in DSL
+func (s *System) Supervisor(id stack.SupervisorID) *Supervisor {
+	super := s.sys.Supervisor(id)
+	return newSupervisor(commonWithLog(s.common, s.log.New("id", id)), super)
+}
 
-func Hydrate(sys stack.System) *System {
+func Hydrate(t stack.T, setup *stack.Setup) *System {
```

Comment:
`setup` contains a `stack.T`

Although something like that may make sense in some cases: I think we should make the typing more clear between test and package level testing.
E.g. by adding a method to the `T` that setup uses that isn't supported by the regular test, and vice-versa, so the setup-cleanup (package-wide) and the test-cleanup (per test) stay distinct. Mixing up the other attributes might also be ugly (failing the whole package instead of just the test).

Reply:
Ah, that probably explains https://github.com/ethereum-optimism/optimism/issues/15143

Reply:
I'm going to try and clean up the package vs test scoped `T` in a follow up - I had a stab it and it winds up affecting a fair bit of stuff (and still doesn't force us to get it right every time which is challenging...).

I am confident that we should create a new DSL per test and so should use the test level T with it which is what we get from `setup` currently so have adjusted this to do that. 

If we spin up a system that should be reused, we need to ensure it uses the package scoped version from orchestrator and that affects a lot of stuff...